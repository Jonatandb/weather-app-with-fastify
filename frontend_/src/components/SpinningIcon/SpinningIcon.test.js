const { render } = require('@testing-library/react')
const { default: SpinningIcon } = require('./SpinningIcon')

describe('SpinningIcon', () => {
  it('renders correctly', () => {
    const { container } = render(<SpinningIcon />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with icon as child', () => {
    const { container } = render(<SpinningIcon>⚡</SpinningIcon>)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with icon as child and speed', () => {
    const { container } = render(<SpinningIcon speed='slow'>⚡</SpinningIcon>)
    expect(container).toMatchSnapshot()
  })
})
