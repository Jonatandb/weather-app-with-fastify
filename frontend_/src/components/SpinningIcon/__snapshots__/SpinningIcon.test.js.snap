// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinningIcon renders correctly 1`] = `
<div>
  <div
    class="SpinningIcon-fast"
  />
</div>
`;

exports[`SpinningIcon renders correctly with icon as child 1`] = `
<div>
  <div
    class="SpinningIcon-fast"
  >
    ⚡
  </div>
</div>
`;

exports[`SpinningIcon renders correctly with icon as child and speed 1`] = `
<div>
  <div
    class="SpinningIcon-slow"
    speed="slow"
  >
    ⚡
  </div>
</div>
`;
